include/glow/Graph/Graph.h

```diff
@@ -261,10 +270,78 @@ class Function final : public Named {
 
   const Module *getParent() const { return parent_; }
 
+  /// Logs the node creation with a list of input nodes.
+  void logNodeCreation(Node *newNode, llvm::ArrayRef<Node *> inputs) {
+    if (dumpCompilationLogOpt) {
+      std::string logStr = "";
+      logStr.append("[FULL SCOPE: ");
+      logStr.append(logCtx_.getFullScopeName());
+      logStr.append("]");
+      logStr.append(" --- CREATE { (");
+      logStr.append(newNode->getKindName());
+      logStr.append(", ");
+      logStr.append(newNode->getName());
+      logStr.append(") <== ");
+      for (auto n : inputs) {
+        logStr.append(" ( ");
+        logStr.append(n->getKindName());
+        logStr.append(", ");
+        logStr.append(n->getName());
+        logStr.append(" ) ");
+      }
+      logStr.append(" }\n");
+      logCtx_.addLogContent(logStr);
+    }
+  }
+
+  /// Logs the node replacement.
+  void logNodeReplacement(Node *oldNode, Node *newNode) {
+    if (dumpCompilationLogOpt) {
+      std::string logStr = "";
```

Comment:
Same comments here about using `glow::strFormat` or `llvm::formatv` and including field names.